baselines/fed_debug/README.md

```diff
@@ -0,0 +1,164 @@
+
+**Models:** This baseline implements two CNN architectures:
+1. ResNet
+2. DenseNet
+3. VGG
+
+**Dataset:** The datasets are partitioned among clients, and each client participates in the training (cross-silo). However, you can easily extend the code to work in cross-device settings. The non-IID is based on quantity-based imbalance ([niid_bench](https://arxiv.org/abs/2102.02079)).
+
+| Dataset | #classes | #clients | partitioning method |
+| :------- | :------: | :------: | :-----------------: |
+| CIFAR-10 | 10 | 30, 50 | IID and Non-IID |
+| FEMNIST | 10 | 30, 50 | IID and Non-IID |
+
+  
+**Training Hyperparameters:**
+
+Default training hyperparameters are in `conf/base.yaml`.
+
+## Environment Setup
+
+Experiments are conducted with `Python 3.11.9`. It is recommended to use Python 3.11 for the experiments.
+
+Check the documentation for the different ways of installing `pyenv`, but one easy way is using the [automatic installer](https://github.com/pyenv/pyenv-installer):
+
+```bash
+curl https://pyenv.run | bash # then, don't forget links to your .bashrc/.zshrc
+```
+
+You can then install any Python version with `pyenv install 3.11.9` Then, in order to use FedDebug baseline, you'd do the following:
+
+```bash
+# cd to your fed_debug directory (i.e. where the `pyproject.toml` is)
+pyenv local 3.11.9
+poetry env use 3.11.9 # set that version for poetry
+
+
+# run this from the same directory as the `pyproject.toml` file is
+poetry install
+poetry shell
+
+# check the python version by running the following command
+python --version # it should be >=3.11.9
+```
+
+This will create a basic Python environment with just Flower and additional packages, including those needed for simulation. Now you are inside your environment (pretty much as when you use `virtualenv` or `conda`).
+
+## Running the Experiments
+
+>Note: You can run almost any evaluation from the paper by changing the parameters in conf/base.yaml. Also, you can change the resources (per client CPU and GPU) in conf/base.yaml to speed up the simulation. Please check flower simulation guide to for more detail ([Flower Framework main](https://flower.ai/docs/framework/how-to-run-simulations.html) ).
+
+```bash
+# This will run the fed_debug with default settings.
+poetry run python -m fed_debug.main 
+```  
+
+
+## Expected Results  
+
+> Note: The following commands will take time to complete.  
+
+```bash
+#To run multiple experiments with different configurations:
+poetry run python -m fed_debug.main --multirun 
+
+# To vary thresholds:
+poetry run python -m fed_debug.main vary_thresholds=True
+
+#To generate graphs and display table. 
+
+poetry run python -m fed_debug.main generate_graphs=True
+```
+
+It will generate a CSV file (fed_debug_results.csv) that you can visualize. Additionally, it will generate a PDF (Figure-10.pdf) containing Figure 10 graphs.
+![Figure 10 of Fed_Debug Paper](Figure-10.png)
```

Comment:
could you include the script to generate the script and instructions people should follow to generate after running the experiments? what experiments should be run exactly?

Reply:
resolved